test/coinselector.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { Coin } from '../src/wallet/coin';
+import { CoinSelector } from '../src/wallet/coin-selector.ts';
+import { Transaction } from 'bitcoinjs-lib';
+import { validTransactions } from './fixtures/transaction';
+
+describe('CoinSelector', () => {
+    let coins: Coin[];
+    let transaction: Transaction;
+    let coinSelector: CoinSelector;
+    const feeRate: number = 5;
+    let totalBalance: number = 0;
+
+    beforeAll(() => {
+        coinSelector = new CoinSelector(feeRate);
+        // Add Balance
+        coins = [];
+        for (let i = 0; i < 10; i++) {
+            const value = Math.random() * 1000;
+            const coin = new Coin({ value });
+            coins.push(coin);
+            totalBalance += coin.value - coin.estimateSpendingFee(feeRate);
+        }
+    });
+
+    it('should generate cost of change', () => {
```

Comment:
For this particular test, I think you should calculate the cost of change by hand and then match it with the returned values.

Reply:
In summary, just match the output of `coinSelector.costOfChange` with the calculated value.